ci/cron/src/Main.hs

```diff
@@ -90,163 +88,90 @@ http_get url = do
-            -- after 0.13.43.
-            if to_v version < to_v "0.13.44"
-            then do
-                shell_ "git -c user.name=CI -c user.email=CI@example.com cherry-pick 0c4f9d7f92c4f2f7e2a75a0d85db02e20cbb497b"
-                build_helper version path
-            else do
-                -- The release-triggering commit does not have a tag, so we
-                -- need to find it by walking through the git history of the
-                -- LATEST file.
-                sha <- find_commit_for_version version
-                Control.Exception.bracket_
-                    (shell_ $ "git checkout " <> sha <> " -- docs/source/support/release-notes.rst")
-                    (shell_ "git reset --hard")
-                    (build_helper version path)
-        build_helper version path = do
-            robustly_download_nix_packages version
+            -- The release-triggering commit does not have a tag, so we need to
+            -- find it by walking through the git history of the LATEST file.
+            sha <- find_commit_for_version version
+            Control.Exception.bracket_
+                (shell_ $ "git checkout " <> sha <> " -- docs/source/support/release-notes.rst")
+                (shell_ "git reset --hard")
+                (build_helper version)
+        build_helper version = do
+            robustly_download_nix_packages
             shell_ $ "DAML_SDK_RELEASE_VERSION=" <> version <> " bazel build //docs:docs"
-            shell_ $ "mkdir -p  " <> path </> version
-            shell_ $ "tar xzf bazel-bin/docs/html.tar.gz --strip-components=1 -C" <> path </> version
-            checksums <- shell $ "cd " <> path </> version <> "; find . -type f -exec sha256sum {} \\; | grep -v 'support/release-notes.html'"
-            writeFile (path </> version </> "checksum") checksums
-        create_versions_json versions path = do
+            shell_ $ "mkdir -p  " <> temp </> version
+            shell_ $ "tar xzf bazel-bin/docs/html.tar.gz --strip-components=1 -C" <> temp </> version
+
+push :: String -> String -> IO ()
+push local remote =
+    shell_ $ "aws s3 cp " <> local <> " " <> "s3://docs-daml-com" </> remote <> " --recursive"
+
+fetch_if_missing :: String -> String -> IO ()
+fetch_if_missing temp folder = do
+    missing <- not <$> Directory.doesDirectoryExist (temp </> folder)
+    if missing then do
+        putStrLn $ "Downloading " <> folder <> "..."
+        shell_ $ "aws s3 cp s3://docs-daml-com" </> folder <> " " <> temp </> folder <> " --recursive"
+        putStrLn "Done."
+    else do
+        putStrLn $ folder <> " already present."
+
+update_s3 :: String -> Versions -> IO ()
+update_s3 temp vs = do
+    putStrLn "Updating versions.json & hidden.json..."
+    create_versions_json (dropdown vs) (temp </> "versions.json")
+    -- order does not matter for hidden ones
+    create_versions_json (Set.toList $ all_versions vs `Set.difference` (Set.fromList $ dropdown vs)) (temp </> "hidden.json")
```

Comment:
Is there a downside to sorting them? While it might not matter it still seems nice to sort things.

Reply:
At this point they're `String`s so it's a bit unwieldy to sort them, and nobody is supposed to look at that file. Not much effort to add, though; I'll look into that.

Reply:
By keeping a `Version` type up to this point it's become fairly easy to sort them, so that's now done.